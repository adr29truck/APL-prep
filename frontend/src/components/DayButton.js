import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';

export const DayButton = (props) => {
  // props
  // object - obj {id, isChecked, style, name}
  // onClickCheckBox - Func

  return (
  <FormControlLabel
    control={
      <Checkbox
        style={ props.object.style }
        name={props.object.name}
        // checked={props.object.isChecked}
        color="primary"
        id={toString(props.object.id)}
        onChange={props.onClickCheckBox}
      />
    }
    label={ props.object.label }
  />)
}