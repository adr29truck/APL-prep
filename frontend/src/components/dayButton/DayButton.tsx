import React from 'react';

import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';

const DayButton = ({ object, onClickCheckBox }: any) => {
  // props
  // object - obj {id, isChecked, style, name}
  // onClickCheckBox - Func

  return (
    <FormControlLabel
      control={
        <Checkbox
          style={object.style}
          name={object.name}
          checked={object.isChecked}
          color="primary"
          id={object.id.toString()}
          onChange={onClickCheckBox}
        />
      }
      label={object.label}
    />
  );
};

export default DayButton;
