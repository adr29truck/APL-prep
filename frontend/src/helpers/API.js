import axios from 'axios';

const baseUrl = () => {
  try {
    return `${window.location.protocol}//${window.location.hostname}:5000/`;
  } catch {
    return 'http://localhost:5000/';
  }
};

/**
 * Represents everythin API related
 */
export default class API {
  /**
   * Does get requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @return {object} JSON response from API request
   */
  static get(route) {
    return axios.get(baseUrl() + route)
        .then((res) => {
          return res.data;
        }).catch((error) => {
          console.log(error);
        });
  }

  /**
   * Does post requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @param {object} data - data to pass to the API
   * @return {object} JSON response from API request
   */
  static post(route, data) {
    return axios.post(baseUrl() + route, data)
        .then((res) => {
          return res.data;
        }).catch((error) => {
          console.log(error);
        });
  }
}
