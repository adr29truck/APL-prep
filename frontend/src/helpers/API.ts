/* eslint-disable no-console */
import axios, { AxiosResponse, AxiosError, AxiosRequestConfig } from 'axios';

const baseUrl = () => {
  try {
    return `${window.location.protocol}//${window.location.hostname}:5000/`;
  } catch {
    return 'http://localhost:5000/';
  }
};

const axiosConfig: AxiosRequestConfig = {
  headers: {
    'content-Type': 'application/json',
    Accept: '/',
    'Cache-Control': 'no-cache',
    // Authorization: localStorage.getItem('jwt'),
    // FIXME: IM BROKEN
    Authorization:
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE2MTEyMzk0NjUsImV4cCI6MTYxMTMyNTg2NSwianRpIjoiYjJhNDY5YTItZGE1My00MGM1LWExNDYtMDg1ZTIxNzBkOTYwIiwiaWQiOjEsInJscyI6IiIsInJmX2V4cCI6MTYxMzgzMTQ2NX0.s702zGzynnTmyBgc0VzB6HJvccIRPMrH75pQ0YcR0fY',
  },
  // credentials: 'true',
};

/**
 * Represents everythin API related
 */
export default class API {
  /**
   * Does get requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @return {object} JSON response from API request
   */
  static get(route: string) {
    return axios
      .get(baseUrl() + route, axiosConfig)
      .then((res: AxiosResponse) => {
        return res.data;
      })
      .catch((error: AxiosError) => {
        console.log(error);
      });
  }

  /**
   * Does post requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @param {object} data - data to pass to the API
   * @return {object} JSON response from API request
   */
  static post(route: string, data: object) {
    return axios
      .post(baseUrl() + route, data, axiosConfig)
      .then((res: AxiosResponse) => {
        return res.data;
      })
      .catch((error: AxiosError) => {
        console.log(error);
      });
  }
}
