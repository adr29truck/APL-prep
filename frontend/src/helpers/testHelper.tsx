import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store';

export const TestHelper = ({ children }: any) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};
