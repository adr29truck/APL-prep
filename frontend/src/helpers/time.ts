/**
 * Returns the selected day as a date string
 * @return {string} - day objects date as string
 */
export function getDayAsString(day: Date) {
  let temp = day;
  if (temp.getTimezoneOffset() < 0) {
    temp = new Date(temp.valueOf() + 60_000 * -temp.getTimezoneOffset());
  } else {
    temp = new Date(temp.valueOf() - 60_000 * temp.getTimezoneOffset());
  }
  return temp.toISOString().split('T')[0];
}
