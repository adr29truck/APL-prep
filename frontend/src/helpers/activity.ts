import API from './API';
import store from '../store';

export async function fetchActivities() {
  const temp = await API.get('activities');
  store.dispatch({ type: 'activity/setActivitiesState', payload: temp });
}
