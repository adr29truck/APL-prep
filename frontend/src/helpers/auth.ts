import store from '../store';
import API from './API';

/**
 * Extracts the content from a JWT token
 * @param {*} token - x
 * @return {Object} - data
 */
export function parseJwt(token: string) {
  const base64Url = token.split('.')[1];
  const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
  const jsonPayload = decodeURIComponent(
    atob(base64)
      .split('')
      .map((c) => {
        return `%${`00${c.charCodeAt(0).toString(16)}`.slice(-2)}`;
      })
      .join('')
  );
  return JSON.parse(jsonPayload);
}

export async function signIn(payload: object) {
  const temp = await API.post('api/login', payload);
  store.dispatch({ type: 'auth/setState', payload: temp });
}

export async function signOut() {
  await API.post('api/logout');
  store.dispatch({ type: 'auth/setState', payload: { id: 0 } });
}
