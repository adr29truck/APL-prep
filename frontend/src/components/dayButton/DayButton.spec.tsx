import React from 'react';
import { mount } from 'enzyme';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import '@testing-library/jest-dom/extend-expect';
import DayButton from './DayButton';
import { TestHelper } from '../../helpers/testHelper';

describe('<DayButton />', () => {
  it('renders correctly', () => {
    let x = false;

    const wrapper = mount(
      <TestHelper>
        <DayButton object={{style: {}, name: 'test', isChecked: false, id: 12, label: 'testlabel'}} onClickCheckBox={ () => x = !x }/>
      </TestHelper>
    );

    expect(wrapper.html()).toMatchSnapshot();
  });
  
  it('handles right click correctly', () => {
    let x = false;
    let func = jest.fn()
    render(
      <TestHelper>
        <DayButton object={{style: {}, name: 'test', isChecked: x, id: 12, label: 'testlabel'}} onClickCheckBox={ func }/>
      </TestHelper>
    );

    expect(screen.getByText('testlabel', {exact: true}))
    userEvent.click(screen.getByText('testlabel'));
    expect(func).toHaveBeenCalledTimes(1);
    // expect(screen.getByLabelText('testlabel').checked).toEqueal(true)
    // userEvent.click(screen.getByTestId('left-arrow'));
  });
});
