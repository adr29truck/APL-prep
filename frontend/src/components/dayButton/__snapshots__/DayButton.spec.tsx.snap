// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DayButton /> renders correctly 1`] = `"<label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input class=\\"PrivateSwitchBase-input-4\\" id=\\"12\\" name=\\"test\\" type=\\"checkbox\\" data-indeterminate=\\"false\\" value=\\"\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\">testlabel</span></label>"`;
